scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -397,16 +368,54 @@ def _output_reader(proc):
             logger.debug('PYTEST: %s', line)
         proc.communicate()
 
-    def _apply_instance_status(self):
-        if self.state:
-            self.instance.status = self.state
-            if self.state in ["error", "failed"]:
-                self.instance.reason = "Pytest failed"
-        else:
-            self.instance.status = "failed"
-            self.instance.reason = "Pytest timeout"
-        if self.instance.status in ["error", "failed"]:
-            self.instance.add_missing_case_status("blocked", self.instance.reason)
+    def update_test_status(self):
+        if not self.state:
+            self.instance.testcases = []
+            try:
+                self._parse_report_file(self.report_file)
+            except Exception as e:
+                logger.error(f'Error when parsing file {self.report_file}: {e}')
```

Comment:
Please use `%` instead of `f-string` in logging message.

Reply:
ditto